server/db.js

```diff
@@ -85,19 +75,135 @@ function init(release) {
                                 // we now run alter table commands for every column that has been added since the early versions
                                 // this is inefficient, but should only run once per upgrade, and will skip over errors of adding columns that already exist
                                 db.serialize(() => {
+                                    // need to make sure all the old source columns are present before we convert to new format
+                                    // probably look at removing this on next release, or making it so it only runs when upgrading from ancient version
                                     db.run("ALTER TABLE capcodes ADD push INTEGER DEFAULT 0", function(err){ /* ignore error */ });
                                     db.run("ALTER TABLE capcodes ADD pushpri TEXT", function(err){ /* ignore error */ });
                                     db.run("ALTER TABLE capcodes ADD pushgroup TEXT", function(err){ /* ignore error */ });
                                     db.run("ALTER TABLE capcodes ADD pushsound TEXT", function(err){ /* ignore error */ });
                                     db.run("ALTER TABLE capcodes ADD mailenable INTEGER DEFAULT 0", function(err){ /* ignore error */ });
+                                    db.run("ALTER TABLE capcodes ADD mailto TEXT", function(err){ /* ignore error */ });
                                     db.run("ALTER TABLE capcodes ADD telegram INTEGER DEFAULT 0", function(err){ /* ignore error */ });
                                     db.run("ALTER TABLE capcodes ADD telechat TEXT", function(err){ /* ignore error */ });
                                     db.run("ALTER TABLE capcodes ADD ignore INTEGER DEFAULT 0", function (err) { /* ignore error */ });
                                     db.run("ALTER TABLE capcodes ADD twitter INTEGER DEFAULT 0", function (err) { /* ignore error */ });
                                     db.run("ALTER TABLE capcodes ADD twitterhashtag TEXT", function (err) { /* ignore error */ });
                                     db.run("ALTER TABLE capcodes ADD discord INTEGER DEFAULT 0", function (err) { /* ignore error */ });
                                     db.run("ALTER TABLE capcodes ADD discwebhook TEXT", function (err) { /* ignore error */ });
-                                    db.run("PRAGMA user_version = "+release, function(err){ /* ignore error */ });
+                                    db.run("ALTER TABLE capcodes ADD pluginconf TEXT", function (err) { /* ignore error */ });
+                                    // begin scary stuff, consider hiding behind a solid object during this bit
```

Comment:
Should probably add a check here to ensure this only runs once. 
I used if (res..user_version < '20181118') when playing with Knex

Reply:
Good catch - will make it so it only executes if upgrading from version prior to this release :)

Reply:
Updated the upgrade scripts, will make this smarter at some point, if not made obsolete by the knex stuff